source/uwp/SharedRenderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -367,314 +371,219 @@ namespace AdaptiveCards::Rendering::Xaml_Rendering
+                            svgImageSource.RasterizePixelWidth(size.Width);
+                        }
 
-                        copyStreamOperation.Completed(
-                            [randomAccessStream, weakThis, imgProperties, image](
-                                auto const& /*operation*/, auto status)
-                            {
-                                if (status == winrt::AsyncStatus::Completed)
-                                {
-                                    randomAccessStream.Seek(0);
-                                    if (auto strongThis = weakThis.get())
-                                    {
-                                        strongThis->HandleAccessStreamForImageSource(
-                                            imgProperties, randomAccessStream, image);
-                                    }
-                                }
-                            });
+                        co_await svgImageSource.SetSourceAsync(stream);
+                    }
+                    else
+                    {
+                        co_await wil::resume_foreground(GetDispatcher(strongImageSource));
+                        auto bitmapImage = strongImageSource.as<winrt::BitmapImage>();
+                        co_await bitmapImage.SetSourceAsync(stream);
                     }
                 }
-            });
-        m_getStreamOperations.push_back(getStreamOperation);
-
-        return image;
-    }
-
-    template<typename TElement, typename TStream>
-    void XamlBuilder::HandleAccessStreamForImageSource(ImageProperties<TElement> const& imgProperties,
-                                                       TStream const& stream,
-                                                       winrt::ImageSource const& imageSource)
-    {
-        SetImageSource(imgProperties.uiElement, imageSource, imgProperties.stretch);
-
-        if (imgProperties.isImageSvg)
-        {
-            // If we have an SVG, we need to try to parse for the image size before setting the image source
-            auto streamSize = stream.Size();
-            auto inputStream = stream.GetInputStreamAt(0);
-            auto streamDataReader = winrt::DataReader(inputStream);
-            auto loadDataReaderOperation = streamDataReader.LoadAsync((uint32_t)streamSize);
-
-            loadDataReaderOperation.Completed(
-                [weakThis = this->get_weak(), streamDataReader, streamRef = winrt::make_weak(stream),
-                 imageSourceRef = winrt::make_weak(imageSource.as<winrt::SvgImageSource>()), imgProperties](
-                    auto const& result, auto status) -> void
+                else
                 {
-                    auto strongThis = weakThis.get();
-                    auto strongImageSource = imageSourceRef.get();
-                    auto strongStream = streamRef.get();
-
-                    if (strongThis && strongImageSource && strongStream)
+                    // Otherwise, no resolver...
```

Comment:
What does this comment refer to? I thought the resolver was used within `ResolveToStream`